tests/ui_tests/test_provider_page.py

```diff
@@ -147,3 +196,125 @@ def test_run_upgrade_on_provider_page_by_toolbar(self, app_fs, sdk_client_fs, up
+        provider_config_page.config.type_in_config_field(
+            row=config_row, value=params["row_value_new"], clear=True, adcm_test=None
+        )
+
+        provider_config_page.config.set_description(params["config_name_new"])
+        provider_config_page.config.save_config()
+        provider_config_page.config.compare_current_to(params["config_name_old"])
+        with allure.step("Check row history"):
+            row_with_history = provider_config_page.config.get_all_config_rows()[0]
+            provider_config_page.config.wait_history_row_with_value(row_with_history, params["row_value_old"])
+
+    def test_reset_config_in_row_on_provider_config_page(self, app_fs, upload_and_create_test_provider):
+        params = {
+            "row_name": "str_param:",
+            "row_value_new": "test",
+            "row_value_old": "0000",
+            "config_name": "test_name",
+        }
+        provider_config_page = ProviderConfigPage(
+            app_fs.driver, app_fs.adcm.url, upload_and_create_test_provider.id
+        ).open()
+        config_row = provider_config_page.config.get_all_config_rows()[0]
+        provider_config_page.config.type_in_config_field(
+            row=config_row, value=params["row_value_new"], clear=True, adcm_test=None
+        )
+        provider_config_page.config.set_description(params["config_name"])
+        provider_config_page.config.save_config()
+
+        provider_config_page.config.reset_to_default(row=config_row)
+        provider_config_page.config.assert_input_value_is(expected_value=params["row_value_new"], row=config_row)
+
+    @pytest.mark.parametrize("bundle", ["provider_required_fields"], indirect=True)
+    def test_field_validation_on_provider_config_page(
+        self, app_fs, sdk_client_fs, bundle, upload_and_create_test_provider
+    ):
+        params = {
+            'pass_name': 'Test password',
+            'req_name': 'Test Required item',
+            'not_req_name': 'Test item',
+            'wrong_value': 'test',
+        }
+        provider_config_page = ProviderConfigPage(
+            app_fs.driver, app_fs.adcm.url, upload_and_create_test_provider.id
+        ).open()
+        provider_config_page.config.check_password_confirm_required(params['pass_name'])
+        provider_config_page.config.check_field_is_required(params['req_name'])
+        config_row = provider_config_page.config.get_all_config_rows()[0]
+        provider_config_page.config.type_in_config_field(params['wrong_value'], row=config_row)
+        provider_config_page.config.check_field_is_invalid(params['not_req_name'])
+
+
+class TestProviderActionPage:
+    @pytest.mark.smoke()
+    def test_open_by_tab_provider_action_page(self, app_fs, upload_and_create_test_provider):
+        provider_main_page = ProviderMainPage(app_fs.driver, app_fs.adcm.url, upload_and_create_test_provider.id).open()
+        provider_main_page.open_actions_tab()
```

Comment:
Maybe make `open_actions_tab` returning `ProviderActionsPage`? For detailed pages, it looks like nice practice to add in our codebase

Reply:
changed